pkg/cloud/services/wait/wait.go

```diff
@@ -69,11 +70,25 @@ func WaitForWithRetryable(backoff wait.Backoff, condition wait.ConditionFunc, re
 		for _, r := range retryableErrors {
 			if code == r {
 				// We should retry.
+				actualErr = err
 				return false, nil
 			}
 		}
 
 		// Got an error that we can't retry, so return it.
 		return false, err
 	})
+
+	// no error, return nil
+	if waitErr == nil {
```

Comment:
I'm wondering if this should also compare against the err returned for a timeout by `wait.ExponentialBackoff` as well? 
```suggestion
	if waitErr == nil || waitErr != wait.ErrWaitTimeout {
```

Reply:
I fixed the errors here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1221